cmd/tnf/addclaim/addclaim.go

```diff
@@ -27,12 +29,205 @@ var (
 		Short: "The test suite generates a \"claim\" file",
 		RunE:  claimUpdate,
 	}
+	claimCompareFiles = &cobra.Command{
+		Use:   "compare",
+		Short: "Compare 2 \"claim\" file",
+		RunE:  claimCompare,
+	}
 )
 
 const (
 	claimFilePermissions = 0o644
 )
 
+type cnistruct []struct {
+	Name    string        "json:\"name\""
+	Plugins []interface{} "json:\"plugins\""
+}
+type Cni struct {
+	Claim struct {
+		Nodes struct {
+			CniPlugins map[string]cnistruct `json:"cniPlugins"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+type Csi struct {
+	Claim struct {
+		Nodes struct {
+			CsiDriver interface{} `json:"csiDriver"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+
+type HwInfo struct {
+	Claim struct {
+		Nodes struct {
+			NodesHwInfo map[string]interface{} `json:"nodesHwInfo"`
+		} `json:"nodes"`
+	} `json:"claim"`
+}
+
+type RawResult struct {
+	Claim struct {
+		RawResults struct {
+			Cnfcertificationtest struct {
+				Testsuites struct {
+					Testsuite struct {
+						Testcase testcase `json:"testcase"`
+					} `json:"testsuite"`
+				} `json:"testsuites"`
+			} `json:"cnf-certification-test"`
+		} `json:"rawResults"`
+	} `json:"claim"`
+}
+
+//nolint:funlen,gocyclo
+func claimCompare(cmd *cobra.Command, args []string) error {
+	var nodes, nodes2 []string
+	claimFileTextPtr := &Claim1
+	dat, err := os.ReadFile(*claimFileTextPtr)
+	if err != nil {
+		log.Fatalf("Error reading claim1 file:%v", err)
+	}
+	claimFileTextPtr2 := &Claim2
+	dat2, err2 := os.ReadFile(*claimFileTextPtr2)
+	if err != nil {
+		log.Fatalf("Error reading claim2 file :%v", err2)
+	}
+	// cniclaimRoot.Claim.RawResults
+	var cni1, cni2 Cni
+	errcni2 := json.Unmarshal(dat2, &cni2)
+	if errcni2 != nil {
+		log.Fatalf("Error in unmarshal the cni from cliam2 file  :%v", errcni2)
```

Comment:
```suggestion
		log.Fatalf("Error in unmarshal the cni from claim2 file  :%v", errcni2)
```

Seems like there's a bunch of these `cliam` to `claim` typos.